var/spack/repos/builtin/packages/py-poetry-plugin-export/package.py

```diff
@@ -15,5 +15,5 @@ class PyPoetryPluginExport(PythonPackage):
     version("1.0.7", sha256="f6ac707ae227b06b2481249ed2678ff6b810b3487cac0fbb66eb0dc2bfd6ecf1")
 
     depends_on("python@3.7:3", type=("build", "run"))
-    depends_on("py-poetry@1.2:1", type=("build", "run"))
+    # depends_on("py-poetry@1.2:1", type=("build", "run")) # circular dependency
```

Comment:
@alalazo FYI another circular dependency, unfortunately at run-time. @pradyunsg how does pip handle this kind of situation?

Reply:
checked in the source code, it is actually a run-time only dependency

Reply:
Can we move this to a separate PR so we can discuss this in more detail without holding up this PR? I'm not sure the best way to handle this.

Reply:
Circular build-time dependencies are not permitted.

Reply:
> Can we move this to a separate PR so we can discuss this in more detail without holding up this PR? I'm not sure the best way to handle this.

Without fixing this, `nf-core-tools` does not even concretise, so on my end I am happy to hold the PR. But happy to do otherwise if you think it is better to completely decouple the two, e.g. for the sake of better managing them.

Reply:
Checked the latest versions of both poetry and poetry-plugin-export but the circular dep still exists. Looks like the developers are aware of this: https://github.com/python-poetry/poetry/pull/5980

I'm not really sure what to do in the meantime. I think the fix you've proposed is about the best we can do. The `py-poetry-plugin-export` package will be broken unless users also install `py-poetry`, but `py-poetry` is probably far more popular and more important to get working.

Reply:
Oh, I forgot to mention one aspect, that makes the current workaround practically quite effective.
In Spack, the only dependent of `py-poetry-plugin-export` is actually `py-poetry`. So if the latter is installed the former will be, too, hence it is unlikely to have a situation where the latter is broken.